qa/workunits/cephadm/test_iscsi_etc_hosts.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+# checks if the container and host's /etc/hosts files match
+# Necessary to avoid potential bugs caused by podman making
+# edits to /etc/hosts file in the container
+# exits with code 1 if host and iscsi container /etc/hosts do no match
+
+set -ex
+
+ISCSI_DAEMON=$(/home/ubuntu/cephtest/cephadm ls | jq -r '.[] | select(.service_name == "iscsi.foo") | .name')
+/home/ubuntu/cephtest/cephadm enter --name $ISCSI_DAEMON -- cat /etc/hosts > iscsi_daemon_etc_hosts.txt
```

Comment:
```
2023-03-22T15:40:39.274 INFO:tasks.workunit:Running workunit cephadm/test_iscsi_etc_hosts.sh...
2023-03-22T15:40:39.275 DEBUG:teuthology.orchestra.run.smithi044:workunit test cephadm/test_iscsi_etc_hosts.sh> mkdir -p -- /home/ubuntu/cephtest/mnt.0/client.0/tmp && cd -- /home/ubuntu/cephtest/mnt.0/client.0/tmp && CEPH_CLI_TEST_DUP_COMMAND=1 CEPH_REF=6f0f906272c8fb053c785bf0340c636123037ba1 TESTDIR="/home/ubuntu/cephtest" CEPH_ARGS="--cluster ceph" CEPH_ID="0" PATH=$PATH:/usr/sbin CEPH_BASE=/home/ubuntu/cephtest/clone.client.0 CEPH_ROOT=/home/ubuntu/cephtest/clone.client.0 CEPH_MNT=/home/ubuntu/cephtest/mnt.0 adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage timeout 3h /home/ubuntu/cephtest/clone.client.0/qa/workunits/cephadm/test_iscsi_etc_hosts.sh
2023-03-22T15:40:39.336 INFO:tasks.workunit.client.0.smithi044.stderr:++ /home/ubuntu/cephtest/cephadm ls
2023-03-22T15:40:39.336 INFO:tasks.workunit.client.0.smithi044.stderr:++ jq -r '.[] | select(.service_name == "iscsi.foo") | .name'
2023-03-22T15:40:39.544 INFO:tasks.workunit.client.0.smithi044.stderr:ERROR: cephadm should be run as root
2023-03-22T15:40:39.553 INFO:journalctl@ceph.mon.a.smithi044.stdout:Mar 22 15:40:39 smithi044 ceph-mon[105578]: pgmap v127: 33 pgs: 33 active+clean; 577 KiB data, 19 MiB used, 268 GiB / 268 GiB avail; 853 B/s rd, 0 op/s
2023-03-22T15:40:39.560 INFO:tasks.workunit.client.0.smithi044.stderr:+ ISCSI_DAEMON=
2023-03-22T15:40:39.560 INFO:tasks.workunit.client.0.smithi044.stderr:+ /home/ubuntu/cephtest/cephadm enter --name -- cat /etc/hosts
2023-03-22T15:40:39.768 INFO:tasks.workunit.client.0.smithi044.stderr:usage: cephadm enter [-h] [--fsid FSID] --name NAME ...
2023-03-22T15:40:39.769 INFO:tasks.workunit.client.0.smithi044.stderr:cephadm enter: error: argument --name/-n: expected one argument
2023-03-22T15:40:39.784 DEBUG:teuthology.orchestra.run:got remote process result: 2
2023-03-22T15:40:39.785 INFO:tasks.workunit:Stopping ['cephadm/test_iscsi_pids_limit.sh', 'cephadm/test_iscsi_etc_hosts.sh'] on client.0...
```
looks like this didn't properly grab the iscsi daemon's name in the test

Reply:
https://pulpito.ceph.com/adking-2023-03-15_01:34:51-orch:cephadm-wip-adk-testing-2023-03-14-1905-distro-default-smithi/7208480

Reply:
Missing `sudo`?